postsRouter.js

```diff
@@ -0,0 +1,129 @@
+const router = require('express').Router();
+const db = require('./data/db');
+
+router.post('/', (req, res) => {
+  const post = req.body;
+  db.insert(post)
+    .then(post => {
+      if (!req.body.title.length || !req.body.contents.length) {
+        res.status(404).json({
+          message: 'Please provide title and contents for the post',
+        });
+      } else {
+        res.status(201).json(post);
+      }
+    })
+    .catch(err => {
+      res.status(500).json({
+        message: 'There was an error while saving the post to the database',
+      });
+    });
+});
+
+router.post('/:id/comments', (req, res) => {
+  let comment = req.body;
+  const id = req.params.id;
+  comment.post_id = id;
+  db.insertComment(comment)
+    .then(post => {
+      if (!comment.text) {
+        res.status(404).json({
```

Comment:
At this point you've already invoked insertComment, this conditional should come before insertComment